matrix_3x3_3.0.cpp

```diff
@@ -0,0 +1,343 @@
+﻿#include <stdio.h>
+#include <stdlib.h>
+#include<time.h>
+
+typedef struct
+{
+	float data[3][3];
+	int width;
+	int height;
+} Matrix;
+
+void print_matrix(Matrix* M)
+{
+	for (int row_num = 0; row_num < M->height; row_num++)
+	{
+		for (int col_num = 0; col_num < M->width; col_num++)
+		{
+			printf("%f\t", M->data[row_num][col_num]);
+		}
+		printf("\n");
+	}
+	printf("\n");
+}
+
+Matrix READ_mat(char* name, int width, int height)
+{
+	Matrix A;
+	A.width = width;
+	A.height = height;
+	FILE* fp;
+	fopen_s(&fp, name, "r");
+	for (int row_num = 0; row_num < A.height; row_num++)
+	{
+		for (int col_num = 0; col_num < A.width; col_num++)
+		{
+			fscanf_s(fp, "%f", &A.data[row_num][col_num]);
+		}
+	}
+	fclose(fp);
+	return A;
+}
+
+void WRITE_to_file(void* A, int operation)
+{
+	FILE* fp;
+
+	if (sizeof(A) == 4)
+	{
+		fopen_s(&fp, "mat_res.txt", "a+");
+		{
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "Det A:");
+			fprintf(fp, "%f", *(float*)A);
+		}
+		fclose(fp);
+	}
+
+	else
+	{
+		switch (operation)
+		{
+		case 1:
+			fopen_s(&fp, "mat_res.txt", "a+");
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "A + B");
+			fclose(fp);
+			break;
+		case 2:
+			fopen_s(&fp, "mat_res.txt", "a+");
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "A x B");
+			fclose(fp);
+			break;
+		case 3:
+			fopen_s(&fp, "mat_res.txt", "a+");
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "transposed A");
+			fclose(fp);
+			break;
+		case 4:
+			fopen_s(&fp, "mat_res.txt", "a+");
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "inverse A");
+			fclose(fp);
+			break;
+		}
+
+		fopen_s(&fp, "mat_res.txt", "a+");
+		fprintf(fp, "%c", "\n");
+		for (int row_num = 0; row_num < ((Matrix*)A)->height; row_num++)
+		{
+			for (int col_num = 0; col_num < ((Matrix*)A)->width; col_num++)
+			{
+				fprintf(fp, "%f\t", ((Matrix*)A)->data[row_num][col_num]);
+			}
+			fprintf(fp, "%c", "\n");
+		}
+		//fprintf(fp, "%c", "\n");
+		fclose(fp);
+	}
+}
+
+Matrix summ(Matrix* A, Matrix* B)

```

Comment:
looks fine. maybe only "summ" can be renamed to something more speaking e.g. matrixSumm ?